app/validation/validator.py

```diff
@@ -585,10 +586,15 @@ def validate_duplicates(self, json_to_validate):
                 block_path = path_list[:path_list.index('blocks') + 2]
 
                 string_path = '/'.join(block_path)
-                # Using a set for items_per_block allows duplicate ids within a specific block.
-                items_per_block[string_path].add(value)
+                # Using a set for unique_ids_per_block allows duplicate ids within a specific block.
+                unique_ids_per_block[string_path].add(value)
+            else:
+                non_block_ids.append(value)
+
+        all_ids = list(unique_ids_per_block.values()) + [[x] for x in non_block_ids]
```

Comment:
I think this would be simpler if we flattened this down to just a list of all ids before checking for duplicates.